compiler/modules/nback/structure.bal

```diff
@@ -67,6 +67,78 @@ final RuntimeFunction mappingConstructFunction = {
 
 const LLVM_INDEX = "i32";
 
+// This is used in the name of the runtime function.
+enum ListReprKind {
+    LIST_GENERIC = "generic",
+    LIST_INT_ARRAY = "int_array",
+    LIST_FLOAT_ARRAY = "float_array"
+}
+
+type ListRepr readonly & object {
+    ListReprKind kind;
+    llvm:Type memberType;
+    int listDescGetIndex;
+    int listDescSetIndex;
+    function buildMember(llvm:Builder builder, Scaffold scaffold, bir:Operand member, t:SemType memberType) returns llvm:Value|BuildError;
+    function buildMemberStore(llvm:Builder builder, Scaffold scaffold, llvm:Value value, bir:Register reg);
+};
+
+readonly class GenericListRepr {
+    *ListRepr;
+    ListReprKind kind = LIST_GENERIC;
+    llvm:Type memberType = LLVM_TAGGED_PTR;
+    int listDescGetIndex = 1;
+    int listDescSetIndex = 2;
+    function buildMember(llvm:Builder builder, Scaffold scaffold, bir:Operand member, t:SemType memberType) returns llvm:Value|BuildError {
+        return buildWideRepr(builder, scaffold, member, REPR_ANY, memberType);
+    }
+    function buildMemberStore(llvm:Builder builder, Scaffold scaffold, llvm:Value value, bir:Register reg) {
+        return buildStoreTagged(builder, scaffold, value, reg);
+    }
+}
+
+readonly class IntListRepr {
```

Comment:
If you are only ever creating one of a class, then it's better to use an object-constructor-expr e.g.

final ListRepr listIntArrayRepr = object {
    ListReprKind kind = LIST_FLOAT_ARRAY
    // etc

};
```